### PR TITLE
Vehicle migration

### PR DESCRIPTION
**Pull request by amichard**
ZEVA-58

This is to let people review and see the upcoming changes to the vehicle table

Changelog:
- Vehicle Table structure changes
- Renaming some of the Models to be more descriptive
- Fixtures for Make, Fuel Types, Vehicle Class and Vehicles based on the NRCAN data

### CODE REVIEW

File: backend/api/models/vehicle_class.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    id = models.CharField(
+        blank=False,
+        db_comment="Vehicle Class Code (e.g. T, I, S, C, M, L)",
+        max_length=3,
+        null=False,
+        primary_key=True
+    )
+
+    class Meta:
+        db_table = 'vehicle_class'
```
Comment (klefler) [inline backend/api/models/vehicle_class.py]: suggested table name = "vehicle_class_type" or "vehicle_class_code"

File: backend/api/models/vehicle_class.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+from django.db import models
+
+from api.models.mixins.effective_dates import EffectiveDates
+from auditable.models import Auditable
+from .mixins.named import Description
+
+
+class VehicleClass(Auditable, Description, EffectiveDates):
+    id = models.CharField(
```
Comment (klefler) [inline backend/api/models/vehicle_class.py]: suggest naming the same as the table name... i.e. vehicle_class_code

File: backend/api/models/vehicle_class.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        db_comment="Vehicle Class Code (e.g. T, I, S, C, M, L)",
+        max_length=3,
+        null=False,
+        primary_key=True
+    )
+
+    class Meta:
+        db_table = 'vehicle_class'
+
+    db_table_comment = "Vehicle Class (Body/Weight class)" \
```
Comment (klefler) [inline backend/api/models/vehicle_class.py]: add "as defined in NRCAN" to the description.

File: backend/api/models/vehicle_fuel_type.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+from django.db import models
+
+from api.models.mixins.effective_dates import EffectiveDates
+from auditable.models import Auditable
+from .mixins.named import Description
+
+
+class FuelType(Auditable, Description, EffectiveDates):
+    id = models.CharField(
```
Comment (klefler) [inline backend/api/models/vehicle_fuel_type.py]: name same as table name 'vehicle_fuel_type'

File: backend/api/models/vehicle_fuel_type.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        db_comment="Fuel type (e.g. B, BX, BZ)",
+        max_length=3,
+        null=False,
+        primary_key=True
+    )
+
+    class Meta:
+        db_table = 'vehicle_fuel_type'
+
+    db_table_comment = "Fuel type of the vehicle" \
```
Comment (klefler) [inline backend/api/models/vehicle_fuel_type.py]: add "as defined in NRCAN" to description.

File: backend/api/models/vehicle.py [hunk 6]
```diff
@@ -30,14 +30,16 @@ class Vehicle(Auditable):
...
+        db_column='vehicle_class_code',
         related_name=None,
         on_delete=models.PROTECT
     )
 
-    type = models.ForeignKey(
-        'Type',
+    fuel_type = models.ForeignKey(
+        'FuelType',
+        db_column='fuel_type_code',
```
Comment (klefler) [inline backend/api/models/vehicle.py]: rename to match pk in parent table, "vehicle_fuel_type'

File: backend/api/models/vehicle.py [hunk 7]
```diff
@@ -53,21 +55,20 @@ class Vehicle(Auditable):
...
-                      max_length=20,
-                      null=False,
-                      default=VehicleDefinitionStates.NEW,
-                      db_comment='The review state of this vehicle. Valid states: {states}'
-                      .format(states=[c.name for c in VehicleDefinitionStates]))
-
-    credit_value = models.OneToOneField(
-        'CreditValue',
-        on_delete=models.CASCADE, null=True
+    state = EnumField(
```
Comment (klefler) [inline backend/api/models/vehicle.py]: can we consider changing the name of the column to 'vehicle_status' or 'vehicle_approval_status' or 'validation_status'

File: backend/api/models/vehicle.py [hunk 8]
```diff
@@ -53,21 +55,20 @@ class Vehicle(Auditable):
...
+        VehicleDefinitionStates,
+        max_length=20,
+        null=False,
+        default=VehicleDefinitionStates.NEW,
+        db_comment="The review state of this vehicle. Valid states: {states}"
```
Comment (klefler) [inline backend/api/models/vehicle.py]: if the column name is changed, we may want to update the comment to "The approval status for whether the vehicle is a valid ZEV, as determined by the Ministry's review. Valid statuses: .........."

**Comment by amichard** [comment]
@klefler I've changed the table names and column names. I'll create a card for changing state to status

**Pull request merged**